<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>turningWindows</class>
 <widget class="QDialog" name="turningWindows">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>775</width>
    <height>514</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTabWidget" name="turningTabWidget">
       <property name="currentIndex">
        <number>2</number>
       </property>
       <widget class="QWidget" name="standard_params">
        <attribute name="title">
         <string>Standard Params</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="basic_pids">
        <attribute name="title">
         <string>Basic PIDs</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_24">
           <item>
            <widget class="QGroupBox" name="gbx_basic_pids_polar">
             <property name="title">
              <string>Polar Axis</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_kp">
                  <property name="text">
                   <string>Kp:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_kp">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.600000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_ki">
                  <property name="text">
                   <string>Ki:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_ki">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.400000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_kd">
                  <property name="text">
                   <string>Kd:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_kd">
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.050000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_i_max">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="baseSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="lineWidth">
                   <number>1</number>
                  </property>
                  <property name="text">
                   <string>Integral Max:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_i_max">
                  <property name="maximum">
                   <double>999.000000000000000</double>
                  </property>
                  <property name="value">
                   <double>30.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_i_dz">
                  <property name="text">
                   <string>Integral DZ:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_i_dz">
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_o_max">
                  <property name="text">
                   <string>Output Max:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_o_max">
                  <property name="value">
                   <double>50.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_15">
                <item>
                 <widget class="QLabel" name="label_basic_pids_polar_lpf_fcut">
                  <property name="text">
                   <string>LPF fCut(Hz): </string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_polar_lpf_fcut"/>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="gbx_basic_pids_pitch">
             <property name="title">
              <string>Pitch Axis</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_kp">
                  <property name="text">
                   <string>Kp:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_kp">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.800000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_9">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_ki">
                  <property name="text">
                   <string>Ki:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_ki">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.400000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_10">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_kd">
                  <property name="text">
                   <string>Kd:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_kd">
                  <property name="value">
                   <double>0.050000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_11">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_i_max">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="baseSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="lineWidth">
                   <number>1</number>
                  </property>
                  <property name="text">
                   <string>Integral Max:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_i_max">
                  <property name="value">
                   <double>30.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_12">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_i_dz">
                  <property name="text">
                   <string>Integral DZ:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_i_dz">
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_13">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_o_max">
                  <property name="text">
                   <string>Output Max:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_o_max">
                  <property name="value">
                   <double>50.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_16">
                <item>
                 <widget class="QLabel" name="label_basic_pids_pitch_lpf_fcut">
                  <property name="text">
                   <string>LPF fCut(Hz): </string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_pitch_lpf_fcut"/>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="gbx_basic_pids_yaw">
             <property name="title">
              <string>Yaw Axis</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_14">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_kp">
                  <property name="text">
                   <string>Kp:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_kp">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.600000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_17">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_ki">
                  <property name="text">
                   <string>Ki:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_ki">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.400000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_18">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_kd">
                  <property name="text">
                   <string>Kd:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_kd">
                  <property name="maximum">
                   <double>9.990000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.050000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_19">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_i_max">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="baseSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="lineWidth">
                   <number>1</number>
                  </property>
                  <property name="text">
                   <string>Integral Max:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_i_max">
                  <property name="value">
                   <double>30.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_20">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_i_dz">
                  <property name="text">
                   <string>Integral DZ:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_i_dz">
                  <property name="value">
                   <double>1.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_21">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_o_max">
                  <property name="text">
                   <string>Output Max:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_o_max">
                  <property name="value">
                   <double>50.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_22">
                <item>
                 <widget class="QLabel" name="label_basic_pids_yaw_lpf_fcut">
                  <property name="text">
                   <string>LPF fCut(Hz): </string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="dsb_basic_pids_yaw_lpf_fcut"/>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QPushButton" name="btn_basic_pids_write_params">
             <property name="text">
              <string>Write Params</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_basic_pids_save_to_flash">
             <property name="text">
              <string>Save to flash</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_basic_pids_refresh_screen">
             <property name="text">
              <string>Refresh Screen</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="fuzzy_pids">
        <attribute name="title">
         <string>Fuzzy PIDs</string>
        </attribute>
        <layout class="QHBoxLayout" name="horizontalLayout_25" stretch="4,6">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QTableView" name="tbv_fuzzy_pids_table">
             <property name="gridStyle">
              <enum>Qt::SolidLine</enum>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_23">
               <item>
                <widget class="QLabel" name="label_fuzzy_pids_items">
                 <property name="text">
                  <string>Items:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="cbox_fuzzy_pids_axis_items">
                 <item>
                  <property name="text">
                   <string>Rx Polar Axis</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Rx Pitch Axis</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Rx Yaw Axis</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="cbox_fuzzy_pids_ctrl_items">
                 <item>
                  <property name="text">
                   <string>Kp</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Ki</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Kd</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="btn_fuzzy_pids_write_params">
               <property name="text">
                <string>Write Params</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_fuzzy_pids_save_to_flash">
               <property name="text">
                <string>Save to flash</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_fuzzy_pids_refresh_screen">
               <property name="text">
                <string>Refresh Screen</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_9" stretch="1,100">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>10</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCustomPlot" name="cpl_fuzzy_pids_graph" native="true"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="adrc">
        <attribute name="title">
         <string>ADRC</string>
        </attribute>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>turningWindows</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>turningWindows</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
